File: Labs/Lab 10/task1.cpp
#include <iostream>
#include <string>
using namespace std;

int main(int argc, char const *argv[])
{
    // Define Variables
    const int ROWS = 4, COLUMNS = 7;
    double weeks[ROWS][COLUMNS];
    double sum = 0, average = 0;
    double averages[4];

    // Prompt user for input
    cout << "Enter the rainfall rate for the four weeks: " << endl;
    for (int i = 0; i < ROWS; i++)
    {
        // Prepare input
        cout << "Week #" << i + 1 << ": ";
        for (int j = 0; j < COLUMNS; j++)
        {
            // Get input
            cin >> weeks[i][j];

            // Add it to sum
            sum += weeks[i][j];

            // Raise average by 1
            average++;
        }
        
        // Calculate average
        average = sum / average;

        // Add average to averages
        averages[i] = average;

        // Reset sum and average
        sum = 0;
        average = 0;
    }

    // Prepare output
    cout << "\n\nClassification for each week: \n";
  
    // Loop through averages
    for (int i = 0; i < ROWS; i++)
    {
        // Rate rainfall based on criteria
        string rating;
        if(averages[i] >= 6)
        {
            rating = "high";
        }
        else if (averages[i] < 6 && averages[i] > 3)
        {
            rating = "Medium";
        }
        else
        {
            rating = "Low";
        }

        // Display output
        cout << "Average Week#" << i + 1 << ": " << rating << " (" << averages[i]  << ")" << endl;
    }

    
    return 0;
}

File: 2022-2023/Term 1/Lab 4/src/Task1.java
import java.util.Scanner;

public class Task1
{
    public static void main(String[] args) throws Exception
    {
        // Start with a scanner
        Scanner in = new Scanner(System.in);

        // Create array 4 weeks 7 days each
        int weeks[][] = new int[4][7];

        // Loop through array and get all values
        for (int i = 0; i < 4; i++)
        {
            System.out.printf("Week %d: ", i + 1);
            for (int j = 0; j < 7; j++)
            {
                weeks[i][j] = in.nextInt();
            }
            System.out.println();
        }

        // Calculate average for each week
        for (int i = 0; i < 4; i++)
        {
            // Prepare Variables
            int sum = 0;
            double average = 0;

            // Calculate average
            for (int j = 0; j < 7; j++)
            {
                sum += weeks[i][j];
            }
            average = sum / 7.0;

            // Print average and classification
            System.out.printf("Week %d: %.2f ", i + 1, average);
            getClassification(average);
        }

        // Close input
        in.close();
    }

    // Get categorization for the rainfall
    public static void getClassification(double average)
    {
        // Check which Category the rainfall goes into
        if (average >= 6)
        {
            System.out.println("(high)");
        }
        else if (average > 3)
        {
            System.out.println("(medium)");
        }
        else
        {
            System.out.println("(low)");
        }
    }
}
